### ISSUE TITLE
/etc/portage/* content parsing doesn't force sorting for ordering of file loading.

### ISSUE THREAD

**Issue by ferringb** [pkgcore/pkgcore#385]
Package.use in particular is affected by this.  Currently [pkgcore/ebuild/domain's `load_property` invokes livefs iter_scan](https://github.com/pkgcore/pkgcore/blob/0517f1315251e5693ef994b2e2ae20d5aef13c6c/src/pkgcore/ebuild/domain.py#L127) to get the list of files; that returns whatever ordering the underlying FS uses.

That in turn can lead to situations where a 00_* package.use files isn't used as the base for files that follow, leading to incorrect USE stacking.

TL;DR: force ordering here.  Portage's logic for this appears to just be a straight up sort, and I can't see any fault w/ doing the same.

**Comment by ferringb** [comment]
As to the fix for this; it should just be a matter of using `sorted_scan`, but some double checking + tests are warranted.
